### PR TITLE
Bump pyright to 1.353

### LINKED ISSUE #6315: Fix pyright errors
PR #6296 bumps `pyright` version from `1.1.336` to `1.1.349` which reports the errors below.

In the meantime, `pyright` bump was ignored: https://github.com/Scille/parsec-cloud/pull/6296#issuecomment-1921441553

```
+ python /home/runner/.cache/pre-commit/pre-commit.pyz run pyright --show-diff-on-failure --verbose --color=always --all-files
Pyright (using `poetry run pyright`).....................................Failed
- hook id: pyright
- duration: 13.1s
- exit code: 1

added 1 package, and audited 2 packages in 2s

found 0 vulnerabilities
/home/runner/work/parsec-cloud/parsec-cloud/server/parsec/asgi/rpc.py
  /home/runner/work/parsec-cloud/parsec-cloud/server/parsec/asgi/rpc.py:91:23 - error: No overloads for "__new__" match the provided arguments (reportCallIssue)
/home/runner/work/parsec-cloud/parsec-cloud/server/parsec/cli/testbed.py
  /home/runner/work/parsec-cloud/parsec-cloud/server/parsec/cli/testbed.py:64:46 - error: "testbed" is possibly unbound (reportPossiblyUnboundVariable)
/home/runner/work/parsec-cloud/parsec-cloud/server/parsec/components/invite.py
  /home/runner/work/parsec-cloud/parsec-cloud/server/parsec/components/invite.py:479:52 - error: Cannot access member "IDLE" for type "type[Annotated]"
    Member "IDLE" is unknown (reportAttributeAccessIssue)
/home/runner/work/parsec-cloud/parsec-cloud/server/parsec/components/memory/organization.py
  /home/runner/work/parsec-cloud/parsec-cloud/server/parsec/components/memory/organization.py:129:15 - error: Method "bootstrap" overrides class "BaseOrganizationComponent" in an incompatible manner
    Parameter 2 name mismatch: base parameter is named "id", override parameter is named "now"
    Parameter 3 name mismatch: base parameter is named "now", override parameter is named "id" (reportIncompatibleMethodOverride)
  /home/runner/work/parsec-cloud/parsec-cloud/server/parsec/components/memory/organization.py:211:15 - error: Method "stats_as_user" overrides class "BaseOrganizationComponent" in an incompatible manner
    Positional parameter count mismatch; base method has 4, but override has 3 (reportIncompatibleMethodOverride)
/home/runner/work/parsec-cloud/parsec-cloud/server/tests/common/client.py
  /home/runner/work/parsec-cloud/parsec-cloud/server/tests/common/client.py:125:15 - error: Method "events_listen" overrides class "BaseAuthenticatedRpcClient" in an incompatible manner
    Return type mismatch: base method returns type "Coroutine[Any, Any, Rep]", override returns type "_AsyncGeneratorContextManager[EventsListenSSE]"
      "_AsyncGeneratorContextManager[EventsListenSSE]" is incompatible with "Coroutine[Any, Any, Rep]" (reportIncompatibleMethodOverride)
6 errors, 0 warnings, 0 informations
```

Full output: https://github.com/Scille/parsec-cloud/actions/runs/7733478793/job/21085652823

### PR DESCRIPTION
**Pull request by mmmarcos**
Closes #6315

### CODE REVIEW

File: server/tests/common/client.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ async def _do_request(self, req: bytes) -> bytes:
         return rep.content
 
     @asynccontextmanager
-    async def events_listen(
+    async def events_listen(  # pyright: ignore [reportIncompatibleMethodOverride]
```
Comment (FirelightFlagboy) [inline server/tests/common/client.py:133]: Why incompatible, can you add a comment above ?

File: server/parsec/components/invite.py [hunk 2]
```diff
@@ -459,7 +459,7 @@ def _event_filter(
                     # Hence it's hazardous to send the event directly from this coroutine (it
                     # requires cancellation shielding), and instead the `send_nowait` will
                     # delegate the work to a dedicated coroutine.
-                    status = InvitationStatusField.IDLE
+                    status = InvitationStatusField.IDLE  # pyright: ignore [reportAttributeAccessIssue]
```
Comment (FirelightFlagboy) [inline server/parsec/components/invite.py:462]: Why access issue, can you add a comment above ?

Comment (mmmarcos) [reply server/parsec/components/invite.py:462]: I don't know why the access issue. It seems that pyright doesn't follow the `Annotated` type, but I couldn't find any reference nor related issue in their repo :confused:

Without the ignore, pyright reports:
```
error: Cannot access member "IDLE" for type "type[Annotated]"
```

File: server/tests/common/client.py (hunk 1 above)
Comment (mmmarcos) [reply server/tests/common/client.py:133]: The return type of `events_listen` here is not compatible with that of the base class `events_listen`:
```
client.py:133:15 - error: Method "events_listen" overrides class "BaseAuthenticatedRpcClient" in an incompatible manner
    Return type mismatch: base method returns type "Coroutine[Any, Any, Rep]", override returns type "_AsyncGeneratorContextManager[EventsListenSSE]"
      "_AsyncGeneratorContextManager[EventsListenSSE]" is incompatible with "Coroutine[Any, Any, Rep]" (reportIncompatibleMethodOverride)
```

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**